html/modules/custom/ghi_plans/src/Plugin/ParagraphHandler/PlanTabSwitcher.php

```diff
@@ -0,0 +1,162 @@
+    if ($plan->getType() !== 'plan' && $plan->getType() !== 'plan_para' && $plan->getType() !== 'plan_entity') {
+      return;
+    }
+
+    // Get parent if needed.
+    if ($plan->getType() === 'plan_entity') {
+      /** @var \Drupal\node\Entity\Node $plan */
+      $plan = $plan->field_plan->first->entity;
+    }
+
+    $overview_link = $this->t('Overview');
+    if (!$plan->isPublished()) {
+      if ($plan->access('view')) {
+        $overview_link = $plan->toLink($overview_link);
+      }
+    }
+    else {
+      $overview_link = $plan->toLink($overview_link);
+    }
+
+    $tabs = [
+      [
+        'title' => $this->t('Overview'),
+        'url' => $plan->toUrl(),
+        'attributes' => [
+          'class' => ['active'],
+        ],
+      ],
+    ];
+
+    // Get all plan entity nodes with plan as parent.
+    $plan_entities = \Drupal::entityTypeManager()->getStorage('node')->loadByProperties([
+      'type' => 'plan_entity',
+      'field_plan_id' => $plan->id(),
+    ]);
+
+    if ($plan_entities) {
+      $suffix = '';
+      if ($config['show_count']) {
+        $suffix = ' <span class="counter">' . count($plan_entities) . '</span>';
+      }
+
+      $tabs[] = [
+        'title' => $this->t('Clusters') . $suffix,
+        'url' => $plan_entities[0]->toUrl(),
+        'attributes' => [
+          'class' => [],
+        ],
+      ];
+    }
+
+    $variables['content'][] = [
+      '#theme' => 'links',
+      '#links' => $tabs,
+      '#attributes' => [
+        'class' => [
+          'links--plan-overview',
+        ],
+      ],
+    ];
+
+    if ($config['render_all_children']) {
+      /** @var \Drupal\hpc_api\Query\EndpointPlanQuery $q */
+      $q = \Drupal::service('hpc_api.endpoint_plan_query');
+      $children = [];
+
+      foreach ($plan_entities as $plan_entity) {
+        if ($plan_entity->field_plan_id && !$plan_entity->field_plan_id->isEmpty()) {
+          $title = $q->getPlanIcon($plan_entity->field_plan_id->value);
+        }
+        else {
+          $title = $plan_entity->label();
+        }
+
+        $children[] = [
+          'title' => $title,
+          'url' => $plan_entity->toUrl(),
+          'attributes' => [
+            'class' => [],
+          ],
+        ];
+      }
+
+      $variables['content'][] = [
+        '#theme' => 'links',
+        '#links' => $children,
+        '#attributes' => [
+          'class' => [
+            'links--plan-entities',
+          ],
+        ],
+      ];
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function widgetAlter(&$element, &$form_state, $context) {
+    parent::widgetAlter($element, $form_state, $context);
+
+    $config = $this->getConfig();
+    $subform = &$element['subform'];
+
+    $subform['show_count'] = [
+      '#title' => $this->t('Show count'),
+      '#type' => 'radios',
+      '#default_value' => $config['show_count'],
+      '#options' => [
+        TRUE => $this->t('Yes'),
+        0 => $this->t('No'),
+      ],
+    ];
```

Comment:
This should be a setting on the plan node that this paragraph is attached to, either directly or via subpages.
For subpages of type `plan_entity` it should follow the setting in `field_plan_hide_ple_counts`, for subpages of type `governing_entity` it should follow the setting in `field_plan_hide_gve_counts`. We don't want this to be manually set for every single subpage to keep consistency.